x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -1021,21 +1020,13 @@ public static Authentication newApiKeyAuthentication(AuthenticationResult<User>
     public Authentication toRemoteAccess(RemoteAccessAuthentication remoteAccessAuthentication) {
         assert isApiKey() : "can only convert API key authentication to remote access";
         assert false == isRunAs() : "remote access does not support authentication with run-as";
+        assert getEffectiveSubject().getUser().roles().length == 0
+            : "the user associated with a remote access authentication must have no role";
         final Map<String, Object> metadata = new HashMap<>(getAuthenticatingSubject().getMetadata());
         final Authentication.RealmRef authenticatedBy = newRemoteAccessRealmRef(getAuthenticatingSubject().getRealm().getNodeName());
-        final User userFromRemoteCluster = remoteAccessAuthentication.getAuthentication().getEffectiveSubject().getUser();
-        assert userFromRemoteCluster.enabled() : "the user received from a remote cluster must be enabled";
-        final User userWithoutRoles = new User(
-            userFromRemoteCluster.principal(),
-            EMPTY_ARRAY,
-            userFromRemoteCluster.fullName(),
-            userFromRemoteCluster.email(),
-            userFromRemoteCluster.metadata(),
-            userFromRemoteCluster.enabled()
-        );
         final Authentication authentication = new Authentication(
             new Subject(
-                userWithoutRoles,
+                getEffectiveSubject().getUser(),
```

Comment:
Did we not have any test that asserts the effective user was the remote user? I don't see any test change that is related to the change here. If we don't have such a test, should we add one now, ie. asserting the effective user of remote access authentication is the API key user?

Reply:
Ya turns out this was only indirectly covered. I'll add an explicit test 👍